tests/test_docs/test_commands.py

```diff
@@ -0,0 +1,181 @@
+
+    if isinstance(cmd, click.Group):
+
+        for sub_cmd_name in cmd.list_commands(click.Context):
+
+            # Get the sub-command
+            sub_cmd = cmd.get_command(click.Context, sub_cmd_name)
+
+            # Recursively build the tree
+            tree["commands"][sub_cmd_name] = get_group_tree(sub_cmd)
+
+    return tree
+
+
+class CommandValidator:
+    """Validates commands against a CLI"""
+
+    def __init__(self, cli: Group):
+        """Extract autonomy command tree from the aea cli"""
+        self.tree = {"commands": {cli.name: get_group_tree(cli)}}
+
+    def validate(self, cmd: str, file_: str = "") -> bool:
+        """Validates a command"""
+
+        # Copy the tree
+        tree = deepcopy(self.tree)
+        latest_subcmd = None
+        allow_option_arg = False
+
+        cmd_parts = [i for i in cmd.split(" ") if i]
+
+        # Iterate the command parts
+        for cmd_part in cmd_parts:
+
+            # Subcommands
+            if cmd_part in tree["commands"].keys():
+                latest_subcmd = cmd_part
+                tree = tree["commands"][cmd_part]
+                allow_option_arg = False
+                continue
+
+            # Options
+            if cmd_part.startswith("-"):
+                if cmd_part not in tree["options"]:
+                    print(
+                        f"Command validation error in {file_}: option '{cmd_part}' is not present on the command tree {list(tree['options'])}:\n    {cmd}"
+                    )
+                    return False
+
+                allow_option_arg = True
+                continue
+
+            # Option arguments: we can't validate them, just guess that they are correct.
+            if allow_option_arg:
+                allow_option_arg = False
+                continue
+
+            # Command arguments
+            if not latest_subcmd:
+                print(
+                    f"Command validation error in {file_}: detected argument '{cmd_part}' but no latest subcommand exists yet:\n    {cmd}"
+                )
+                return False
+
+            if not tree["arguments"]:
+                print(
+                    f"Command validation error in {file_}: argument '{cmd_part}' is not valid as the latest subcommand [{latest_subcmd}] does not admit arguments:\n    {cmd}"
+                )
+                return False
+
+            # If we reach here, this command part is probably an argument for a command.
+            # We can't validate it, just guess that it is correct.
+
+        return True
+
+
+def test_validate_doc_commands() -> None:
+    """Test that doc commands are valid"""
+
+    # Get the markdown files and the Makefile
+    target_files = list(Path("docs").rglob("*.md"))
+    target_files.append(Path("Makefile"))
+
+    # Get the validator
+    validator = CommandValidator(aea_cli)
+
+    COMMAND_REGEX = r"""(^|\s|`|>)(?P<full_cmd>(?P<cli>aea) ((?!(&|'|\(|\[|\n|\.|`|\||#|<\/code>|=|")).)*)"""
```

Comment:
I've now made this regex a little bit more strict. If this gets merged, I'll open another PR to match this in open-autonomy.

Reply:
it's hard to understand this regex and I can't verify correctness.